src/main/java/org/prebid/server/handler/NotificationEventHandler.java

```diff
@@ -0,0 +1,148 @@
+            respondWithBadStatus(context, ex.getMessage());
+            return;
+        }
+
+        final NotificationEvent notificationEvent = new NotificationEvent(eventNotificationRequest.getType(),
+                eventNotificationRequest.getBidId(), eventNotificationRequest.getBidder());
+
+        analyticsReporter.processEvent(notificationEvent);
+
+        final Map<String, String> queryParams = HttpContext.from(context).getQueryParams();
+        final String format = queryParams.get(FORMAT_PARAMETER);
+
+        try {
+            validateEventRequestQueryParams(format);
+        } catch (InvalidRequestException ex) {
+            respondWithBadStatus(context, ex.getMessage());
+            return;
+        }
+
+        respondWithOkStatus(context, format);
+    }
+
+    private EventNotificationRequest makeEventRequest(Buffer body) {
+        if (body == null) {
+            throw new InvalidRequestException(
+                    "Request body was empty. Expected request with body has next fields: type, bidid and bidder.");
+        }
+
+        final EventNotificationRequest eventNotificationRequest;
+        try {
+            eventNotificationRequest = Json.decodeValue(body, EventNotificationRequest.class);
+        } catch (DecodeException e) {
+            throw new InvalidRequestException(
+                    "Request body couldn't be parsed. Expected request body has next fields: type, bidid and bidder.");
+        }
+
+        validateEventRequestBody(eventNotificationRequest);
+        return eventNotificationRequest;
+    }
+
+    private void validateEventRequestBody(EventNotificationRequest eventNotificationRequest) {
+        final String type = eventNotificationRequest.getType();
+        if (ObjectUtils.notEqual(type, VIEW_TYPE) && ObjectUtils.notEqual(type, WIN_TYPE)) {
+            throw new InvalidRequestException(String.format(
+                    "Type is required parameter. Possible values are win and view, but was %s", type));
+        }
+
+        final String bidId = eventNotificationRequest.getBidId();
+        if (StringUtils.isBlank(bidId)) {
+            throw new InvalidRequestException("bidid is required and can't be empty.");
+        }
+
+        final String bidder = eventNotificationRequest.getBidder();
+        if (StringUtils.isBlank(bidder)) {
+            throw new InvalidRequestException("bidder is required and can't be empty.");
+        }
+
+    }
+
+    private void validateEventRequestQueryParams(String format) {
+        if (format != null && ObjectUtils.notEqual(format, JPG) && ObjectUtils.notEqual(format, PNG)) {
+            throw new InvalidRequestException("format when defined should has value of png or jpg.");
+        }
+    }
+
+    private void respondWithBadStatus(RoutingContext context, String message) {
+        context.response().setStatusCode(HttpResponseStatus.BAD_REQUEST.code())
+                .end(String.format("%s%s", "Request is invalid: ", message));
```

Comment:
`String.format("Request is invalid: %s", message)` looks more readable, isn't it?